src/Routing/RouteCollection.php

```diff
@@ -155,10 +155,6 @@ public function parseRequest(ServerRequestInterface $request): array
                 if ($r === null) {
                     continue;
                 }
-                if ($uri->getQuery()) {
-                    parse_str($uri->getQuery(), $queryParameters);
-                    $r['?'] = array_merge($r['?'] ?? [], $queryParameters);
-                }
```

Comment:
I think we originally implemented this to support webservers that are no longer used often now that phpfpm and modphp have wide adoption and newer servers like frankenphp and caddy don't have the same problems.